plugins/base/src/main/kotlin/transformers/pages/samples/SamplesTransformer.kt

```diff
@@ -55,19 +56,35 @@ abstract class SamplesTransformer(val context: DokkaContext) : PageTransformer {
         }
     }.toMap()
 
-    private fun ContentNode.addSample(contentPage: ContentPage, platform: PlatformData, fqName: String, analysis: Map<PlatformData, EnvironmentAndFacade>): ContentNode {
-        val facade = analysis[platform]?.facade ?:
-            return this.also { context.logger.warn("Cannot resolve facade for platform ${platform.name}")}
+    private fun ContentNode.addSample(contentPage: ContentPage, platform: PlatformData?, fqName: String, analysis: Map<PlatformData, EnvironmentAndFacade>): ContentNode {
+        val facade = if(platform == null) {
+            analysis.entries.find { it.key.platformType.name == "common" }?.value
+        } else {
+            analysis[platform]
+        }?.facade ?: return this.also { context.logger.warn("Cannot resolve facade for platform ${platform?.name ?: "expect"}") }
         val psiElement = fqNameToPsiElement(facade, fqName) ?:
             return this.also { context.logger.warn("Cannot find PsiElement corresponding to $fqName") }
         val imports = processImports(psiElement) // TODO: Process somehow imports. Maybe just attach them at the top of each body
         val body = processBody(psiElement)
-        val node = platformHintedContentCode(platform, contentPage.dri, body, "kotlin")
-        return this.safeAs<ContentGroup>()?.run { copy(
-            children = children.indexOfFirst { contentNode ->
-                contentNode.safeAs<ContentHeader>()?.children?.firstOrNull()?.safeAs<ContentText>()?.text == "Sample"
-            }.takeIf { it != -1 }?.let { children.apply { this.safeAs<MutableList<ContentNode>>()?.add(it+1, node) } } ?: children.also { context.logger.warn("Not found Sample block in ${contentPage.dri}")}
-        ) } ?: this.also { context.logger.warn("ContentPage ${contentPage.dri} cannot be cast to ContentGroup") }
+        val node = contentCode(contentPage.platforms(), contentPage.dri, body, "kotlin")
```

Comment:
You can at least inline `val body` or both of them

Reply:
Is is likely that proper samples presentation requires further manipulations on imports/body. I think inlining won't change much.